packages/ui-common/src/components/table.tsx

```diff
@@ -0,0 +1,152 @@
+import { Component, For, createSignal, JSX } from 'solid-js';
+import { FaSolidSortDown } from 'solid-icons/fa';
+import { FaSolidSortUp } from 'solid-icons/fa';
+import { FaSolidSort } from 'solid-icons/fa';
+import { SortOptions } from '../interfaces/sort-options.interface';
+
+type ItemBase = Record<string, any> & { id: string };
+
+export interface Column<Item extends ItemBase> {
+  key: string;
+  title: string;
+  sortable?: boolean;
+  render?: (item: Item) => JSX.Element;
+}
+
+export interface TableProps<Item extends ItemBase> {
+  columns: Column<Item>[];
+  data: Item[];
+  onSort: (options: SortOptions) => void;
+  actions?: {
+    icon: Component | string;
+    label: string;
+    props?: (item: Item) => Record<string, any>;
+    handler: (item: Item) => void;
+  }[];
+}
+
+export const CastmillTable = <Item extends ItemBase>(
+  props: TableProps<Item>
+): JSX.Element => {
+  const [sortConfig, setSortConfig] = createSignal({
+    key: undefined,
+    direction: 'ascending',
+  } as SortOptions);
+  const [selectedRows, setSelectedRows] = createSignal(new Set());
+
+  const handleSort = async (key: string) => {
+    const direction =
+      sortConfig().key === key && sortConfig().direction === 'ascending'
+        ? 'descending'
+        : 'ascending';
+    setSortConfig({ key, direction });
+    props.onSort({ key, direction });
+  };
+
+  const sortIcon = (column: Column<any>): JSX.Element => {
+    if (sortConfig() && sortConfig().key === column.key) {
+      return sortConfig().direction === 'ascending' ? (
+        <FaSolidSortUp />
+      ) : (
+        <FaSolidSortDown />
+      );
+    }
+    return column.sortable ? <FaSolidSort /> : <></>; // Empty fragment for non-sortable columns
+  };
+
+  const handleSelectRow = (id: string, isChecked: boolean) => {
+    setSelectedRows((prevSelected) => {
+      const newSet = new Set(prevSelected);
+      if (isChecked) {
+        newSet.add(id);
+      } else {
+        newSet.delete(id);
+      }
+      return newSet;
+    });
+  };
+
+  const handleSelectAll = (event: Event) => {
+    const target = event.target as HTMLInputElement;
+    const newSet = new Set<string>();
+    if (target.checked) {
+      for (const item of props.data) {
+        newSet.add(item.id);
+      }
+    }
+    setSelectedRows(newSet);
+  };
+
+  return (
+    <table>
```

Comment:
In the future we might want to use grid instead of table. It may simplify reactiveness and stuff.